ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/capella/block/BlockProcessorCapella.java

```diff
@@ -170,48 +170,35 @@ public void processBlsToExecutionChangesNoValidation(
     }
   }
 
-  public void processWithdrawals(
-      final MutableBeaconState genericState, final Optional<ExecutionPayload> maybePayload)
-      throws BlockProcessingException {
-    final ExecutionPayload payload =
-        maybePayload.orElseThrow(
-            () ->
-                new BlockProcessingException(
-                    "ExecutionPayload was not found during block processing."));
-    processWithdrawals(genericState, payload);
-  }
-
   @Override
   public void processWithdrawals(
-      final MutableBeaconState genericState, final ExecutionPayload payload)
+      final MutableBeaconState genericState, final ExecutionPayloadSummary payloadSummary)
       throws BlockProcessingException {
     final MutableBeaconStateCapella state = MutableBeaconStateCapella.required(genericState);
-    final ExecutionPayloadCapella executionPayloadCapella =
-        ExecutionPayloadCapella.required(payload);
-    final SszList<Withdrawal> payloadWithdrawals = executionPayloadCapella.getWithdrawals();
-    final List<Withdrawal> expectedWithdrawals = getExpectedWithdrawals(state);
-    if (expectedWithdrawals.size() != payloadWithdrawals.size()) {
+    final SszList<Withdrawal> expectedWithdrawals =
+        schemaDefinitionsCapella
+            .getExecutionPayloadSchema()
+            .getWithdrawalsSchemaRequired()
+            .createFromElements(getExpectedWithdrawals(state));
+
+    if (payloadSummary.getOptionalWithdrawalsRoot().isPresent()
```

Comment:
We should fail if withdrawals root isn't present as that means withdrawals were missing (and/or we've got a block from the wrong milestone somehow).